client/src/pages/Messages/MessagesDashboard.tsx

```diff
@@ -0,0 +1,250 @@
+import { useEffect, useState, useRef } from 'react';
+import {
+  Box,
+  Typography,
+  Grid,
+  List,
+  ListItem,
+  ListItemIcon,
+  Avatar,
+  ListItemText,
+  Divider,
+  Paper,
+  Button,
+  TextField,
+  Modal,
+} from '@mui/material';
+import PageContainer from '../../components/PageContainer/PageContainer';
+import { useAuth } from '../../context/useAuthContext';
+import ConversationInterface from '../../interface/Conversation';
+import getConversations from '../../helpers/APICalls/Messages/getConversations';
+import { useSnackBar } from '../../context/useSnackbarContext';
+import moment from 'moment';
+import sendMessage from '../../helpers/APICalls/Messages/sendMessage';
+import CircularProgress from '@mui/material/CircularProgress';
+import Message from './Message/Message';
+import SendIcon from '@mui/icons-material/Send';
+import CreateConversation from './CreateConversation';
+import useStyles from './useStyles';
+
+type Conversations = ConversationInterface[];
+
+export default function MessagesDashboard(): JSX.Element {
+  const classes = useStyles();
+
+  const [conversations, setConversations] = useState<Conversations>([]);
+  const [isSubmitting, setSubmitting] = useState(false);
+  const { updateSnackBarMessage } = useSnackBar();
+  const [chatbox, setChatBox] = useState<number>(0);
+  const { loggedInUser } = useAuth();
+  const [message, setMessage] = useState<string>('');
+  const scrollRef = useRef<HTMLLIElement>(null);
+  const [open, setOpen] = useState(false);
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setMessage(event.target.value);
+  };
+
+  useEffect(() => {
+    if (scrollRef.current) {
+      scrollRef.current.scrollIntoView();
+    }
+  }, [chatbox]);
+
+  const handleSubmit = (description: string, conversationId: string) => {
+    setSubmitting(true);
+    sendMessage(description, conversationId).then((data) => {
+      if (data.error) {
+        setSubmitting(false);
+        updateSnackBarMessage(data.error.message);
+      } else if (data.success) {
+        console.log(data.success);
+        updateSnackBarMessage('Sent!');
+        getConversations().then((data) => {
```

Comment:
I don't think we need to be refetching all the conversations each time we send a message